src/StateManager.ts

```diff
@@ -1991,6 +1994,97 @@ export default class StateManager {
     this._stage.draw();
   }
 
+  public static isValidAutomaton(
+    json: SerializableAutomaton,
+  ): [boolean, string] {
+    // for now just console log, not sure how to desiplay window
+    if (
+      !json ||
+      !json.states ||
+      !json.alphabet ||
+      !json.transitions ||
+      !json.startState ||
+      !json.acceptStates
+    ) {
+      console.error("Missing required fields");
+      return [false, "Missing required fields"];
+    }
+
+    if (!json.states.every((state) => this.isValidState(state))) {
+      console.error("states not properly formatted");
+      return [false, "states not properly formatted"];
```

Comment:
```suggestion
      return [false, "This automaton's states could not be read."];
```
Since this text is user-facing, we should make sure its grammar is correct and written in full sentences.